### ISSUE TITLE
Ruby reserved words in filters

### ISSUE THREAD

**Issue by nebulon42** [systemed/glug#4]
I'm using a datasource (http://osm2vectortiles.org), which makes extensive use of the column name "class". Using this in filters leads to errors. I think this has something to do with "class" being a reserved word. But also writing `on('class'=='residential')` yields the following error:

````
	from /var/lib/gems/1.9.1/gems/glug-0.0.2/lib/glug.rb:244:in `_set_filter'
	from /var/lib/gems/1.9.1/gems/glug-0.0.2/lib/glug.rb:200:in `on'
	from (eval):11:in `block (2 levels) in <main>'
	from /var/lib/gems/1.9.1/gems/glug-0.0.2/lib/glug.rb:89:in `instance_eval'
	from /var/lib/gems/1.9.1/gems/glug-0.0.2/lib/glug.rb:89:in `layer'
	from (eval):5:in `block in <main>'
	from ../glug/bin/glug:5:in `instance_eval'
	from ../glug/bin/glug:5:in `block in <main>'
	from /var/lib/gems/1.9.1/gems/glug-0.0.2/lib/glug.rb:71:in `instance_eval'
	from /var/lib/gems/1.9.1/gems/glug-0.0.2/lib/glug.rb:71:in `initialize'
	from ../glug/bin/glug:5:in `new'
	from ../glug/bin/glug:5:in `<main>'
````

Is there another way of using reserved words I'm missing or does something in the code of Glug need changing? @systemed

**Comment by nebulon42** [comment]
By looking at the code I was not able to find out how the arguments to the `on()` function are parsed. It seems that when the first part of the filter expression is either a variable, a reserved word or a string this evaluates to `false` otherwise to a proper `Condition`.

**Comment by systemed** [comment]
Glug uses `method_missing` to interpret conditions. As you've spotted, this won't work where the 'key' is a Ruby reserved word and therefore not `missing`!

As a workaround, you can define the key as an object explicitly:

    _class = Glug::OSMKey.new('class')

and then refer to it from then onwards like this:

    on(_class=='residential')

Going forward, I think the best way to solve this would be a short-form version of the above, where you could write `tag('class')=='residential'` where a reserved word was needed.

**Comment by nebulon42** [comment]
Thanks for the explanation! Your workaround did it for me.

**Comment by nebulon42** [comment]
Fixed in 0a20718963ae74420588d539a789d9541bfa2bc3.
Can now write e.g. `on(tag('class')=='residential')`.

**Comment by systemed** [comment]
Advance warning: this will change to `get('class')` in the next version of Glug, for consistency with MBGL expressions.
